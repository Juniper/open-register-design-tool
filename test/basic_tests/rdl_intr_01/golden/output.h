//   Ordt 230719.01 autogenerated file 
//   Input: ./rdl_intr_01/test.rdl
//   Parms: ./rdl_intr_01/test.parms
//   Date: Thu Jul 20 13:46:48 EDT 2023
//

#ifndef __FOO_REGISTER_MAP__
#define __FOO_REGISTER_MAP__

#include <bits.h>

/* FOO_REGISTERS memory map */
enum FOO_REGS {
  BAR_A_REG_0 = 0x0,
  BAR_A_REG_1 = 0x4,
  BAR_INTR_REG1 = 0x100,
  BAR_LOG_REG1 = 0x104,
  BAR_LOG_REG2 = 0x108,
  BAR_LOG_REG3 = 0x10c,
  BAR_SUB_INTR_REG2 = 0x200,
  BAR_SUB_INTR_REG3 = 0x300,
  BAR_RST_REG = 0x304,
  BAR_NEGINTR = 0x500,
  BAR_BOTHINTR = 0x504,
  INTR_CASCADE_MERGE = 0x1000
};

/* BAR_A_REG_0 register fields */
#define BAR_A_REG_0_FLD1 GENMASK(9, 0)
#define BAR_A_REG_0_FLD2 BIT(15)

/* BAR_A_REG_1 register fields */
#define BAR_A_REG_1_FLD1 GENMASK(9, 0)
#define BAR_A_REG_1_FLD2 BIT(15)

/* BAR_INTR_REG1 register fields */
#define BAR_INTR_REG1_INT1 BIT(0)    /* interrupt 1 */
#define BAR_INTR_REG1_INT2 BIT(1)    /* interrupt 2 */

/* BAR_LOG_REG1 register fields */
#define BAR_LOG_REG1_ERR_LOG GENMASK(9, 0)    /* error log field */

/* BAR_LOG_REG2 register fields */
#define BAR_LOG_REG2_ERR_LOG GENMASK(9, 0)    /* error log field */

/* BAR_LOG_REG3 register fields */
#define BAR_LOG_REG3_ERR_LOG GENMASK(9, 0)    /* error log field */

/* BAR_SUB_INTR_REG2 register fields */
#define BAR_SUB_INTR_REG2_INT1 BIT(0)    /* interrupt 1 */
#define BAR_SUB_INTR_REG2_INT2 BIT(1)    /* interrupt 2 */

/* BAR_SUB_INTR_REG3 register fields */
#define BAR_SUB_INTR_REG3_INT1 BIT(0)    /* interrupt 1 */
#define BAR_SUB_INTR_REG3_INT2 BIT(1)    /* interrupt 2 */

/* BAR_RST_REG register fields */
#define BAR_RST_REG_FLD1 GENMASK(7, 4)
#define BAR_RST_REG_FLD2 GENMASK(3, 0)
#define BAR_RST_REG_FLD3 GENMASK(10, 8)
#define BAR_RST_REG_FLD4 GENMASK(13, 11)

/* BAR_NEGINTR (Reg negedge intr) register fields */
#define BAR_NEGINTR_INTERRUPT BIT(0)    /* negedge_intr check */

/* BAR_BOTHINTR (Reg bothedge intr) register fields */
#define BAR_BOTHINTR_INTERRUPT BIT(0)    /* bothedge_intr check */

/* INTR_CASCADE_MERGE register fields */
#define INTR_CASCADE_MERGE_FLD1 BIT(0)
#define INTR_CASCADE_MERGE_FLD2 BIT(1)

#endif
