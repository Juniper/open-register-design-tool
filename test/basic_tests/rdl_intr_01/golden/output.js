/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_intr_01/test.rdl
 *   Parms: ./rdl_intr_01/test.parms
 *   Date: Sun Nov 17 17:01:01 EST 2019
 */

register_set foo "Registers for foo" {

  address = 0x0;
  register_width = 32;
  register_set bar "bar register_set" {
    address = 0x0;
    
    register a_reg "a_reg register" {
      address = 0x0;
      repeat = 2;
      nop[16];
      integer fld2[1] "fld2 field" param {
        access_mode = READ_WRITE;
        reset = unknown;
      };
      nop[5];
      integer fld1[10] "fld1 field" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register intr_reg1 "intr_reg1 register" {
      address = 0x100;
      nop[30];
      integer int2[1] "interrupt 2" param {
        access_mode = WRITE_ONE_TO_CLEAR;
        reset = 0x0;
      };
      integer int1[1] "interrupt 1" param {
        access_mode = WRITE_ONE_TO_CLEAR;
        reset = 0x0;
      };
    };
    
    register log_reg1 "log_reg1 register" {
      address = 0x104;
      nop[22];
      integer err_log[10] "error log field" param {
        description = "{
          this field will capture the incoming value of the hw write bus while intr_reg1.int1 is inactive
        }";
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register log_reg2 "log_reg2 register" {
      address = 0x108;
      nop[22];
      integer err_log[10] "error log field" param {
        description = "{
          this field will capture the incoming value of the hw write bus once when intr_reg1.int1 fires
        }";
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register log_reg3 "log_reg3 register" {
      address = 0x10c;
      nop[22];
      integer err_log[10] "error log field" param {
        description = "{
          this field will capture the incoming value of the hw write bus while all interrupts in intr_reg1 are inactive
        }";
        access_mode = READ_WRITE;
        reset = 0x0;
      };
    };
    
    register_set sub "sub register_set" {
      address = 0x200;
      
      register intr_reg2 "intr_reg2 register" {
        address = 0x0;
        nop[30];
        integer int2[1] "interrupt 2" param {
          access_mode = WRITE_ONE_TO_CLEAR;
          reset = 0x0;
        };
        integer int1[1] "interrupt 1" param {
          access_mode = WRITE_ONE_TO_CLEAR;
          reset = 0x0;
        };
      };
      
      register intr_reg3 "intr_reg3 register" {
        address = 0x100;
        nop[30];
        integer int2[1] "interrupt 2" param {
          access_mode = WRITE_ONE_TO_CLEAR;
          reset = 0x0;
        };
        integer int1[1] "interrupt 1" param {
          access_mode = WRITE_ONE_TO_CLEAR;
          reset = 0x0;
        };
      };
      
      register_set_size = 0x104;
    };
    
    register rst_reg "rst_reg register" {
      address = 0x304;
      nop[18];
      integer fld4[3] "fld4 field" param {
        access_mode = READ_WRITE;
        reset = 0x4;
      };
      integer fld3[3] "fld3 field" param {
        access_mode = READ_WRITE;
        reset = 0x3;
      };
      integer fld1[4] "fld1 field" param {
        access_mode = READ_WRITE;
        reset = 0x1;
      };
      integer fld2[4] "fld2 field" param {
        access_mode = READ_WRITE;
        reset = 0x2;
      };
    };
    
    register negintr "Reg negedge intr" {
      address = 0x500;
      description = "{
        A register with negedge intr
      }";
      nop[31];
      integer interrupt[1] "negedge_intr check" param {
        description = "{
          checking that the negedge intr works
        }";
        access_mode = WRITE_ONE_TO_CLEAR;
        reset = 0x0;
      };
    };
    
    register bothintr "Reg bothedge intr" {
      address = 0x504;
      description = "{
        A register with bothedge intr
      }";
      nop[31];
      integer interrupt[1] "bothedge_intr check" param {
        description = "{
          checking that the bothedge intr works
        }";
        access_mode = WRITE_ONE_TO_CLEAR;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x508;
  };
  
  register_set intr_cascade "intr_cascade register_set" {
    address = 0x1000;
    
    register merge "merge register" {
      address = 0x0;
      access_mode = READ_ONLY;
      nop[30];
      integer fld2[1] "fld2 field" param {
        access_mode = READ_ONLY;
        reset = 0x0;
      };
      integer fld1[1] "fld1 field" param {
        access_mode = READ_ONLY;
        reset = 0x0;
      };
    };
    
    register_set_size = 0x4;
  };
  
  register_set_size = 0x1004;
};

