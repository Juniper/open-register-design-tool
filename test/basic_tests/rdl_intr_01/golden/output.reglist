//   Ordt 190725.01 autogenerated file 
//   Input: ./rdl_intr_01/test.rdl
//   Parms: ./rdl_intr_01/test.parms
//   Date: Sun Nov 17 17:02:53 EST 2019
//

0x0              size: 32   fields: 2    bits: 11        bar.a_reg_0
0x4              size: 32   fields: 2    bits: 11        bar.a_reg_1
0x100            size: 32   fields: 2    bits: 2         bar.intr_reg1
0x104            size: 32   fields: 1    bits: 10        bar.log_reg1
0x108            size: 32   fields: 1    bits: 10        bar.log_reg2
0x10c            size: 32   fields: 1    bits: 10        bar.log_reg3
0x200            size: 32   fields: 2    bits: 2         bar.sub.intr_reg2
0x300            size: 32   fields: 2    bits: 2         bar.sub.intr_reg3
0x304            size: 32   fields: 4    bits: 14        bar.rst_reg
0x500            size: 32   fields: 1    bits: 1         bar.negintr
0x504            size: 32   fields: 1    bits: 1         bar.bothintr
0x1000           size: 32   fields: 2    bits: 2         intr_cascade.merge
