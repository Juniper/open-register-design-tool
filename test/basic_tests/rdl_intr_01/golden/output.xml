<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 191120.01 autogenerated file 
       Input: ./rdl_intr_01/test.rdl
       Parms: ./rdl_intr_01/test.parms
       Date: Wed Nov 20 22:19:48 EST 2019
 -->

<map version="191120.01">
  <id>foo</id>
  <baseaddr>0x0</baseaddr>
  <shorttext>foo registers</shorttext>
  <regset>
    <id>bar</id>
    <shorttext>bar registers</shorttext>
    <baseaddr>0x0</baseaddr>
    <reg>
      <id>a_reg</id>
      <parentpath>bar</parentpath>
      <shorttext>a_reg register</shorttext>
      <baseaddr>0x0</baseaddr>
      <width>32</width>
      <reps>2</reps>
      <stride>0x4</stride>
      <field>
        <id>fld1</id>
        <shorttext>fld1 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>rw</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
      </field>
      <field>
        <id>fld2</id>
        <shorttext>fld2 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <lowidx>15</lowidx>
        <width>1</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>intr_reg1</id>
      <parentpath>bar</parentpath>
      <shorttext>intr_reg1 register</shorttext>
      <baseaddr>0x100</baseaddr>
      <width>32</width>
      <field>
        <id>int1</id>
        <shorttext>interrupt 1</shorttext>
        <access>W1C</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
      </field>
      <field>
        <id>int2</id>
        <shorttext>interrupt 2</shorttext>
        <access>W1C</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
      </field>
      <access>W1C</access>
    </reg>
    <reg>
      <id>log_reg1</id>
      <parentpath>bar</parentpath>
      <shorttext>log_reg1 register</shorttext>
      <baseaddr>0x104</baseaddr>
      <width>32</width>
      <field>
        <id>err_log</id>
        <shorttext>error log field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <wel></wel>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
        <longtext><![CDATA[this field will capture the incoming value of the hw write bus while intr_reg1.int1 is inactive]]></longtext>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>log_reg2</id>
      <parentpath>bar</parentpath>
      <shorttext>log_reg2 register</shorttext>
      <baseaddr>0x108</baseaddr>
      <width>32</width>
      <field>
        <id>err_log</id>
        <shorttext>error log field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <we></we>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
        <longtext><![CDATA[this field will capture the incoming value of the hw write bus once when intr_reg1.int1 fires]]></longtext>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>log_reg3</id>
      <parentpath>bar</parentpath>
      <shorttext>log_reg3 register</shorttext>
      <baseaddr>0x10c</baseaddr>
      <width>32</width>
      <field>
        <id>err_log</id>
        <shorttext>error log field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>w</hwaccess>
          <wel></wel>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>10</width>
        <hwmod></hwmod>
        <longtext><![CDATA[this field will capture the incoming value of the hw write bus while all interrupts in intr_reg1 are inactive]]></longtext>
      </field>
      <access>RW</access>
    </reg>
    <regset>
      <id>sub</id>
      <parentpath>bar</parentpath>
      <shorttext>sub registers</shorttext>
      <baseaddr>0x200</baseaddr>
      <reg>
        <id>intr_reg2</id>
        <parentpath>bar.sub</parentpath>
        <shorttext>intr_reg2 register</shorttext>
        <baseaddr>0x200</baseaddr>
        <width>32</width>
        <field>
          <id>int1</id>
          <shorttext>interrupt 1</shorttext>
          <access>W1C</access>
          <hwinfo>
            <hwaccess>na</hwaccess>
          </hwinfo>
          <reset>0x0</reset>
          <lowidx>0</lowidx>
          <width>1</width>
          <hwmod></hwmod>
          <intr>
            <type>LEVEL</type>
            <stickytype>STICKYBIT</stickytype>
            <enable>intr_reg1.int1</enable>
          </intr>
        </field>
        <field>
          <id>int2</id>
          <shorttext>interrupt 2</shorttext>
          <access>W1C</access>
          <hwinfo>
            <hwaccess>na</hwaccess>
          </hwinfo>
          <reset>0x0</reset>
          <lowidx>1</lowidx>
          <width>1</width>
          <hwmod></hwmod>
          <intr>
            <type>LEVEL</type>
            <stickytype>STICKYBIT</stickytype>
            <maskintrbits></maskintrbits>
            <enable>intr_reg1.int2</enable>
          </intr>
        </field>
        <access>W1C</access>
      </reg>
      <reg>
        <id>intr_reg3</id>
        <parentpath>bar.sub</parentpath>
        <shorttext>intr_reg3 register</shorttext>
        <baseaddr>0x300</baseaddr>
        <width>32</width>
        <field>
          <id>int1</id>
          <shorttext>interrupt 1</shorttext>
          <access>W1C</access>
          <hwinfo>
            <hwaccess>na</hwaccess>
          </hwinfo>
          <reset>0x0</reset>
          <lowidx>0</lowidx>
          <width>1</width>
          <hwmod></hwmod>
          <intr>
            <type>LEVEL</type>
            <stickytype>STICKYBIT</stickytype>
            <enable>sub.intr_reg2.int1</enable>
          </intr>
        </field>
        <field>
          <id>int2</id>
          <shorttext>interrupt 2</shorttext>
          <access>W1C</access>
          <hwinfo>
            <hwaccess>na</hwaccess>
          </hwinfo>
          <reset>0x0</reset>
          <lowidx>1</lowidx>
          <width>1</width>
          <hwmod></hwmod>
          <intr>
            <type>LEVEL</type>
            <stickytype>STICKYBIT</stickytype>
            <maskintrbits></maskintrbits>
            <enable>sub.intr_reg2.int2</enable>
          </intr>
        </field>
        <access>W1C</access>
      </reg>
      <highaddr>0x303</highaddr>
    </regset>
    <reg>
      <id>rst_reg</id>
      <parentpath>bar</parentpath>
      <shorttext>rst_reg register</shorttext>
      <baseaddr>0x304</baseaddr>
      <width>32</width>
      <field>
        <id>fld1</id>
        <shorttext>fld1 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x1</reset>
        <lowidx>4</lowidx>
        <width>4</width>
      </field>
      <field>
        <id>fld2</id>
        <shorttext>fld2 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x2</reset>
        <lowidx>0</lowidx>
        <width>4</width>
      </field>
      <field>
        <id>fld3</id>
        <shorttext>fld3 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x3</reset>
        <lowidx>8</lowidx>
        <width>3</width>
      </field>
      <field>
        <id>fld4</id>
        <shorttext>fld4 field</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0x4</reset>
        <lowidx>11</lowidx>
        <width>3</width>
      </field>
      <access>RW</access>
    </reg>
    <reg>
      <id>negintr</id>
      <parentpath>bar</parentpath>
      <shorttext>Reg negedge intr</shorttext>
      <baseaddr>0x500</baseaddr>
      <width>32</width>
      <longtext><![CDATA[A register with negedge intr]]></longtext>
      <field>
        <id>interrupt</id>
        <shorttext>negedge_intr check</shorttext>
        <access>W1C</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>NEGEDGE</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
        <longtext><![CDATA[checking that the negedge intr works]]></longtext>
      </field>
      <access>W1C</access>
    </reg>
    <reg>
      <id>bothintr</id>
      <parentpath>bar</parentpath>
      <shorttext>Reg bothedge intr</shorttext>
      <baseaddr>0x504</baseaddr>
      <width>32</width>
      <longtext><![CDATA[A register with bothedge intr]]></longtext>
      <field>
        <id>interrupt</id>
        <shorttext>bothedge_intr check</shorttext>
        <access>W1C</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>BOTHEDGE</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
        <longtext><![CDATA[checking that the bothedge intr works]]></longtext>
      </field>
      <access>W1C</access>
    </reg>
    <highaddr>0x507</highaddr>
  </regset>
  <regset>
    <id>intr_cascade</id>
    <shorttext>intr_cascade registers</shorttext>
    <baseaddr>0x1000</baseaddr>
    <reg>
      <id>merge</id>
      <parentpath>intr_cascade</parentpath>
      <shorttext>merge register</shorttext>
      <baseaddr>0x1000</baseaddr>
      <width>32</width>
      <field>
        <id>fld1</id>
        <shorttext>fld1 field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
          <nextassign>bar.intr_reg1(intr)</nextassign>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>0</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
          <input>bar.intr_reg1</input>
        </intr>
      </field>
      <field>
        <id>fld2</id>
        <shorttext>fld2 field</shorttext>
        <access>RO</access>
        <hwinfo>
          <hwaccess>na</hwaccess>
        </hwinfo>
        <reset>0x0</reset>
        <lowidx>1</lowidx>
        <width>1</width>
        <hwmod></hwmod>
        <intr>
          <type>LEVEL</type>
          <stickytype>STICKYBIT</stickytype>
        </intr>
      </field>
      <access>RO</access>
    </reg>
    <highaddr>0x1003</highaddr>
  </regset>
</map>
